components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -610,21 +613,39 @@ private Response handleFormPostResponseModeError(OAuthMessage oAuthMessage, Stri
 
     private Response handleDenyConsent(OAuthMessage oAuthMessage) throws OAuthSystemException, URISyntaxException {
 
+
         OAuth2Parameters oauth2Params = getOauth2Params(oAuthMessage);
         OpenIDConnectUserRPStore.getInstance().putUserRPToStore(getLoggedInUser(oAuthMessage),
-                getOauth2Params(oAuthMessage).getApplicationName(),
-                false, oauth2Params.getClientId());
-        // return an error if user denied
-        OAuthProblemException ex = OAuthProblemException.error(OAuth2ErrorCodes.ACCESS_DENIED,
-                "User denied the consent");
-        String denyResponse = EndpointUtil.getErrorRedirectURL(oAuthMessage.getRequest(), ex, oauth2Params);
+                getOauth2Params(oAuthMessage).getApplicationName(), false, oauth2Params.getClientId());
+
+        OAuthErrorDTO oAuthErrorDTO = EndpointUtil.getOAuth2Service().handleUserConsentDenial(oauth2Params);
+        OAuthProblemException ex = buildConsentDenialException(oAuthErrorDTO);
```

Comment:
Better to have a proper variable name instead of ex? WDYT?

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1701